src/interpreter/mod.rs

```diff
@@ -822,14 +802,14 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         }
     }
 
-    fn eval_lvalue(&mut self, lvalue: &mir::Lvalue<'tcx>) -> EvalResult<'tcx, Lvalue> {
+    fn eval_lvalue(&mut self, lvalue: &mir::Lvalue<'tcx>) -> EvalResult<'tcx, Value> {
```

Comment:
actually, I messed up the one time I used `Value::ByVal(PrimVal::Ptr(ptr))`, that should not have happened. `Value::ByVal(PrimVal::Ptr(ptr))` has a completely different meaning from `Value::ByRef(ptr)`. It's just the `ByValPair` that I abused.

But I see your point and I'll revert to Lvalue